scripts/install_apt_deps.sh

```diff
@@ -1,21 +1,209 @@
 #!/bin/bash
 
+
+### Initial setup ###
+
+# Source environment variables describing the current OS distribution
 . /etc/os-release
 
-sudo apt install -y software-properties-common curl gnupg2
-curl https://apt.kitware.com/keys/kitware-archive-latest.asc | sudo apt-key add -
-sudo apt-add-repository -u -y "deb https://apt.kitware.com/ubuntu/ ${UBUNTU_CODENAME} main"
-sudo apt-key adv --keyserver keys.gnupg.net --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE || sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-key F6E65AC044F831AC80A06380C8B3A55A6F3EFCDE
-sudo add-apt-repository "deb http://realsense-hw-public.s3.amazonaws.com/Debian/apt-repo bionic main" -u
+
+### Helper functions ###
+
+# Function for printing a warning message (with red foreground)
+function print_warning() { echo -e "\e[31m*Warning* ${1}\e[39m"; }
+
+# Function for flattening a string with items seperated by new lines
+# and white space to a single line seperated by a single space.
+# When calling flatten list, the multi-line list (string) passed as
+# the first argument must be protected by double quotes when expanding
+# a variable.
+# Good sample call site: flat_string=$(flatten_list "${multi_line_string}")
+# Bad sample call site:  flat_string="$(flatten_list ${multi_line_string})"
+function flatten_list() { echo "${1}" | xargs; }
```

Comment:
Just wondering, do you think it's cleaner to use Bash arrays like:
```
array=(
    elem1
    elem2
)
...
sudo apt-get install -y "${array[@]}"
```

Reply:
Good call! Always forget about bash arrays. :P

Reply:
Unfortunately, it seems that the real issue was passing in the delimiter separated list to a bash function, resulting in the same gross workaround at call sites even when using bash array. I did keep the bash arrays though, since it was still a bit cleaner :P.